pkg/util/csi/volume_snapshot_test.go

```diff
@@ -951,86 +952,143 @@ func TestGetVolumeSnapshotClass(t *testing.T) {
+	}
+
+	// storageclasses
+	fooStorageClass := &storagev1api.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "foo",
+			Labels: map[string]string{},
+		},
+		Provisioner: "foo.csi.k8s.io",
+	}
+
+	foolabelStorageClass := &storagev1api.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+			Labels: map[string]string{
+				"velero.io/csi-volumesnapshot-class": "sctest",
+			},
+		},
+		Provisioner: "foo.csi.k8s.io",
+	}
+
+	barStorageClass := &storagev1api.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "bar",
+			Labels: map[string]string{},
+		},
+		Provisioner: "bar.csi.k8s.io",
+	}
+
+	hostpathStorageClass := &storagev1api.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "hostpath",
+			Labels: map[string]string{},
+		},
+		Provisioner: "hostpath.csi.k8s.io",
+	}
+
+	blahStorageClass := &storagev1api.StorageClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "blah",
+			Labels: map[string]string{},
+		},
+		Provisioner: "blah.csi.k8s.io",
+	}
+
 	objs := []runtime.Object{hostpathClass, fooClass, barClass, fooClassWithoutLabel, barClass2}
 	fakeClient := velerotest.NewFakeControllerRuntimeClient(t, objs...)
 
 	testCases := []struct {
-		name        string
-		driverName  string
-		pvc         *v1.PersistentVolumeClaim
-		backup      *velerov1api.Backup
-		expectedVSC *snapshotv1api.VolumeSnapshotClass
-		expectError bool
+		name         string
+		storageClass *storagev1api.StorageClass
+		pvc          *v1.PersistentVolumeClaim
+		backup       *velerov1api.Backup
+		expectedVSC  *snapshotv1api.VolumeSnapshotClass
+		expectError  bool
 	}{
 		{
-			name:        "no annotations on pvc and backup, should find hostpath volumesnapshotclass using default behavior of labels",
-			driverName:  "hostpath.csi.k8s.io",
-			pvc:         pvcNone,
-			backup:      backupNone,
-			expectedVSC: hostpathClass,
-			expectError: false,
+			name:         "no annotations on pvc and backup, should find hostpath volumesnapshotclass using default behavior of labels",
+			storageClass: hostpathStorageClass,
+			pvc:          pvcNone,
+			backup:       backupNone,
+			expectedVSC:  hostpathClass,
+			expectError:  false,
 		},
 		{
-			name:        "foowithoutlabel VSC annotations on pvc",
-			driverName:  "foo.csi.k8s.io",
-			pvc:         pvcFoo,
-			backup:      backupNone,
-			expectedVSC: fooClassWithoutLabel,
-			expectError: false,
+			name:         "foowithoutlabel VSC annotations on pvc",
+			storageClass: fooStorageClass,
+			pvc:          pvcFoo,
+			backup:       backupNone,
+			expectedVSC:  fooClassWithoutLabel,
+			expectError:  false,
 		},
 		{
-			name:        "foowithoutlabel VSC annotations on pvc, but csi driver does not match, no annotation on backup so fallback to default behavior of labels",
-			driverName:  "bar.csi.k8s.io",
-			pvc:         pvcFoo,
-			backup:      backupNone,
-			expectedVSC: barClass,
-			expectError: false,
+			name:         "footwithoutlabel VSC annotations on StorageClass",
```

Comment:
```suggestion
			name:         "foowithoutlabel VSC annotations on StorageClass",
```

Reply:
actually.. isn't this the test case where storage class used has the new label with this PR's new functionality?

Reply:
Yes it is, I forgot to update the name. Thanks for catching.